### ISSUE TITLE
imagemin-jpeg-recompress returns Error when compressing

### ISSUE THREAD

**Issue by marocas** [danielgtaylor/jpeg-archive#14]
´´´´
$ npm install imagemin-jpeg-recompress
npm WARN package.json BY_Mailer@0.0.1 No repository field.
|
> jpeg-recompress-bin@2.0.2 postinstall C:\sites\_testes\node_modules\imagemin-j
peg-recompress\node_modules\jpeg-recompress-bin
> node lib/install.js

  √ jpeg-recompress pre-build test passed successfully

imagemin-jpeg-recompress@4.0.0 node_modules\imagemin-jpeg-recompress
├── is-jpg@1.0.0
├── through2@0.6.3 (xtend@4.0.0, readable-stream@1.0.33)
└── jpeg-recompress-bin@2.0.2 (logalot@2.0.0, bin-wrapper@2.1.1)

C:\>grunt imagemin
Running "imagemin:main" (imagemin) task
Warning: Error: Premature end of JPEG file
Bogus Huffman table definition
 in file _images/mee.jpg Use --force to continue.

Aborted due to warnings.
´´´´

my task config is this:

´´´´
var imageCompressEngine = require('imagemin-jpeg-recompress');

  var globalConfig = {

    imagemin: {
      main: {
        options: {
          optimizationLevel: 7,
          pngquant: true,
          progressive: true,
          use: [imageCompressEngine()]
        },
        files: [
          {
            expand: true,
            cwd: '<%= cfg.root %><%= cfg.src.images %>',
            src: ['**/*.{png,jpg,jpeg,gif,ico}'],
            dest: '<%= cfg.root %><%= cfg.dest.images %>'
          }
        ]
      }
    },
});
´´´´
My dependences are the latest versions of 
  "grunt-contrib-imagemin": "^0.9.2", and "imagemin-jpeg-recompress": "^4.0.0".

I'm working on windows OS, if i use another plugin, like imagemin-mozjpeg, it works perfectly.
I want to use this plugin, because it as a better compression.

Best regards, Mário Silva

**Comment by Landen13** [comment]
I have the same issue.

There are 2 jpgs, saved via Photoshop.

The first image was saved through Save As option (not optimized).

Error:
```
Error: Premature end of JPEG file
JPEG datastream contains no image
```

The second image was saved through Save For Web (progressive: true, quality: 61)

Error:
```
Error: Premature end of JPEG file
Bogus DQT index 9
```

**Comment by Landen13** [comment]
Just checked: jpeg-recompress.exe DIDN'T throw any error, while compressing.

**Comment by danielgtaylor** [comment]
@Landen13 do you have a link to a sample file? The link above is dead. I'd like to reproduce this problem locally.

**Comment by Landen13** [comment]
Yeah, here -> http://www39.zippyshare.com/v/he5mvWb6/file.html
This one was saved in Photoshop via Save As (basic/standard, quality 12/12)

**Comment by CoolOppo** [comment]
Dude WTF! Don't go around posting adware sites. I went to that link on my phone, which is secured with the latest Chrome Beta version, and that opened up three popups, not to mention the ads all over the screen.

**Comment by Landen13** [comment]
Lol. Ok, here is the direct link > http://impressivesound.net/01.jpg

Sorry for that. I work with zippyshare from my desktop with Adblock, so there were no problems.

**Comment by danielgtaylor** [comment]
@Landen13 I have downloaded your sample image and then done the following on Windows and it worked perfectly:

```bash
$ jpeg-recompress.exe 01.jpg test.jpg
Metadata size is 21kb
ssim at q=67 (40 - 95): 0.997217
ssim at q=81 (68 - 95): 0.998685
ssim at q=88 (82 - 95): 0.999433
ssim at q=92 (89 - 95): 0.999759
ssim at q=94 (93 - 95): 0.999848
Final optimized ssim at q=95: 0.999810
New size is 60% of original (saved 35 kb)
```

The output is identical on Mac and Linux. Can you try that command and see if it is still broken for you? If the command works, but the imagemin plugin fails, I would open an issue with the author of the plugin because it doesn't seem like it's a Windows issue.

**Comment by shinnn** [comment]
I've commented: https://github.com/imagemin/imagemin-jpeg-recompress/issues/12#issuecomment-91639380

Updating jpeg-recomress binary which imagemin plugin depends on fixes the problem. It doesn't use the latest version (2.1.0) now.

**Comment by Landen13** [comment]
Daniel, i've already checked the command-line jpeg-recompress, and there were no problems.
But imagemin plugin fails, even the updated one v4.2.0 with 2.1.0.

**Comment by shinnn** [comment]
@danielgtaylor As we discussed in https://github.com/imagemin/imagemin-jpeg-recompress/issues/12, Windows version of jpeg-recompress binary emits the error @Landen13 is running into, when it's used with stdin/stdout. (Please check @drazik's comment. https://github.com/imagemin/imagemin-jpeg-recompress/issues/12#issuecomment-93419590)

**Comment by danielgtaylor** [comment]
@shinnn @Landen13 thanks, I'll look into the stdin/stdout issue on Windows and see what I can come up with.

**Comment by danielgtaylor** [comment]
@shinnn @Landen13 please try the build artifact from here, which should include the stdin/stdout fix:

https://ci.appveyor.com/project/danielgtaylor/jpeg-archive/build/16/artifacts

If you can confirm then I can push a new patch release in the near future.

**Comment by shinnn** [comment]
Sorry I'm not an actual Windows user.

@Landen13 @drazik Can you help us?

**Comment by drazik** [comment]
Can't try it before tomorrow. I'll let you know asap.

**Comment by Landen13** [comment]
It worked!
I tryed jpeg-recompress.exe without -s parameter. The output image file is valid!

**Comment by drazik** [comment]
Yep it worked for me too. That's great :) Thank you !

**Comment by Landen13** [comment]
@danielgtaylor, Would you update jpeg-archive with this fix? Gulp plugin imagemin-jpeg-recompress doesn't work now on Windows, and it would be updated only after the new release of jpeg-archive.

**Comment by danielgtaylor** [comment]
Sorry I'm traveling on vacation at the moment but will push a new release
soon.

**Comment by danielgtaylor** [comment]
This is now released with version 2.1.1. Thanks for the help debugging the problem!
